src/main/java/org/craftercms/studio/api/v2/utils/GitRepositoryHelper.java

```diff
@@ -651,6 +658,37 @@ private boolean checkoutSandboxBranch(String site, Repository sandboxRepo, Strin
         }
     }
 
+    private boolean removePublishBlackList(Repository publishedRepo) {
+        String blacklistConfig = studioConfiguration.getProperty(CONFIGURATION_PUBLISHING_BLACKLIST_REGEX);
+        if (!StringUtils.isNotEmpty(blacklistConfig)) {
+            return true;
+        }
+
+        List<String> patterns = Arrays.asList(StringUtils.split(blacklistConfig, STRING_SEPARATOR));
+        try (Git git = new Git(publishedRepo)) {
+            String rootPath = publishedRepo.getWorkTree().getPath();
+            RmCommand rmCommand = git.rm();
+            Files.walkFileTree(Paths.get(rootPath), new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+                    String sitePath = file.toAbsolutePath().toString().replaceFirst(rootPath, FILE_SEPARATOR);
+                    boolean isMatched = ContentUtils.matchesPatterns(sitePath, patterns);
+                    if (isMatched) {
+                        String gitPath = getGitPath(sitePath);
+                        rmCommand.addFilepattern(gitPath);
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+
+            retryingRepositoryOperationFacade.call(rmCommand);
```

Comment:
@phuongnq don't we need to add & commit the files after executing the rm?

Reply:
Hi @jdrossl Thanks for the comment. 
It seems like with or without a commit command, the whole files will be added to an "Initial publish." commit anyway. 
But I added a commit command for consistency. Note that when checking out to `live` branch, this commit ID will be merged to an initial commit.

Reply:
Let's not create an empty commit if it's not needed then, please rollback that change. Thanks

Reply:
Hi @jdrossl The commit is not empty but it will be replaced with an initial commit in `live` branch. I think we can keep it. I will send you a video for the detail

Reply:
Hi @phuongnq, if that commit is not completely necessary (the changes will be included in the next commit) I think we should avoid it for performance reasons (in large sites each commit can take a long time to complete)

Reply:
Hi @jdrossl Got it. I removed the commit. Note that blacklist files are only removed in `live` branch. If you switch to `mater` or sandbox branch, it is still there (since we don't commit the change). But I guess this is OK as we only need to remove it in `live`.
![diff_live_and_test](https://user-images.githubusercontent.com/2996543/170164686-db9abcb0-3bbf-4ab2-9eef-cf8bbf8b6ff0.png)

Reply:
yeah, I think they have always been in the other branches